torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4493,9 +4493,19 @@ def _maybe_evaluate_static(
             #       shape_0 = sympy.Symbol("shape_0", positive=True, integer=True)
             #       expr = sympy.Eq(shape_0 - 1/3, 4)
             #       expr.xreplace({}) # False
-            offset = int(lower - 1)
+
+            # Compute the greatest integer smaller than lower
+            if lower >= 0:
+                lower_integral = int(lower)
+            else:
+                lower_integral = int(lower) - 1
+            # k_vr is the value range we wish for k
+            k_vr = ValueRanges(lower_integral, vr.upper)
```

Comment:
I am looking at one of the test failures. It is happening because at this line ```lower_integral > vr.upper```. Stepping into the debugger, it seems like I am in the unhappy situation where a size-like symbol has a value range ```[0, 1]``` but guard_size_oblivious is attempting to set the lower bound to 2.

I studied the code above and discovered this:
```
                # This is a bit dodgy: what this means is that there was a
                # size-like unbacked symbol whose upper bound < 2.  This
                # causes... problems.
                if lower <= vr.upper:
                    vr = ValueRanges(lower, vr.upper)
```
Sounds like the code author is worried about the same thing and that is why the update of vr is being gated.

Reply:
I fixed the problem using  a similar dodge.